### DISCUSSION TITLE
v21.0.0-beta1

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3935] [Releases]
🚀 **OpenMage 21 is the next step in the evolution of our beloved M1 platform.** 🚀

📢🔔 **While V21 introduces breaking changes that require careful consideration during the upgrade process, these long-overdue modifications will ultimately bring significant benefits to your stores. In the long run, you can expect enhancements in performance, security, SEO scores, and more, making the transition worthwhile.** 📢🔔

## Changes you must know and that you have to check one by one
* [PHP8.2 is now the minimum required version for OM v21](https://github.com/OpenMage/magento-lts/pull/3920)
* [Added form key validation to Contacts form](https://github.com/OpenMage/magento-lts/pull/3146)  
Check `app/design/frontend/base/default/template/contacts/form.phtml` in your theme and be sure that you've `<?php echo $this->getBlockHtml('formkey') ?>` under the `<form` opening tag, [as shown here](https://github.com/OpenMage/magento-lts/pull/3146/files#diff-24566c45709e3e8559774b20e9258080a456999c3b02bef20feda95151a7f8df)
* [Unified CSRF configuration](https://github.com/OpenMage/magento-lts/pull/3147)  
If you enabled CSRF in `System -> Configuration -> System` (which you should!) then it will be enabled for all your forms and the checkout
* [Removed double span element from HTML buttons](https://github.com/OpenMage/magento-lts/pull/3123)  
Be sure to check your CSS styles for the buttons in your frontend theme, you may need to adjust them slightly
* [Removed all deprecated `_Mysql4_` classes](https://github.com/OpenMage/magento-lts/pull/2730)  
This will break many extremely old modules but it was long overdue and allowed us to remove 500+ obsolete files from the core. We provide the `shell/rename-mysql4-class-to-resource.php` migration script to fix the old modules, you will have to run it but make sure you'll do it in a test environment and thoroughly test everything before publishing to production
* [Removed `decorate*()` js functions](https://github.com/OpenMage/magento-lts/pull/3410)  
Check the phtml files of your custom themes for javascript code calling `decorate*()` (decorateList, decorateTable, decorateGeneric etc) and remove those calls since they will generate javascript errors on your browser's console
* [Removed "admin routing compatibility mode" for extensions](https://github.com/OpenMage/magento-lts/pull/1551)
The new admin routing mode was implemented in Magento 1.4 for security reasons, together with "compatibility mode" (in order to allow some time to developers to update their modules). After these many years we are removing the compatibility mode and all extensions will have to follow the new admin routing mode or they will not appear in the backend.
* [Removed `scriptaculous/dragdrop.js` from frontend](https://github.com/OpenMage/magento-lts/pull/3215)

# Changes you want to know about
* RWD theme: updated jQuery to 3.7.1 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3922

# Changes that probably won't affect you
* RWD theme: removed `enquire.js` and converted to window.matchMedia by @fballiano in https://github.com/OpenMage/magento-lts/pull/3208
* RWD theme: converted default logos to SVG (and removed some Magento names) by @fballiano in https://github.com/OpenMage/magento-lts/pull/3148
* Removed onmouseover/onmouseout from adminhtml menu by @luigifab in https://github.com/OpenMage/magento-lts/pull/2737
* Fixed implementation of SessionHandlerInterface in Mage_Core_Model_Resource_Session by @fballiano in https://github.com/OpenMage/magento-lts/pull/3499
* Rewrote `js/varien/weee.js` without prototypejs by @fballiano in https://github.com/OpenMage/magento-lts/pull/3670
* Rewrote `js/varien/accordion.js` without prototypejs by @fballiano in https://github.com/OpenMage/magento-lts/pull/3669
* Rewrote `js/mage/captcha.jsp without prototypejs by @fballiano in https://github.com/OpenMage/magento-lts/pull/3754
* Removed unused file `js/mage/adminhtml/scrollbar.js` by @fballiano in https://github.com/OpenMage/magento-lts/pull/3765
* Removed unused file `js/mage/adminhtml/magento-all.js` by @fballiano in https://github.com/OpenMage/magento-lts/pull/3764
* Removed unused file `js/mage/adminhtml/image.js` by @fballiano in https://github.com/OpenMage/magento-lts/pull/3758
* Removed deprecated and unused files from lib/Mage by @fballiano in https://github.com/OpenMage/magento-lts/pull/3872
* Rewrote `js/mage/adminhtml/hash.js` using native javascript by @fballiano in https://github.com/OpenMage/magento-lts/pull/3763
* Rewrote `js/mage/adminhtml/variables.js` without prototypejs by @fballiano in https://github.com/OpenMage/magento-lts/pull/3762
* Rewrote `js/mage/translate.js` without prototypejs by @fballiano in https://github.com/OpenMage/magento-lts/pull/3662

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v21.0.0-beta1'>v21.0.0-beta1</a>.</em>

**Comment by fballiano** [comment]
Today I upgraded one of my customers' store to v21-beta1 (on a staging environment obviously) and, from first tests, works great!!

The upgrade was truly effortless, just ran shell/rename-mysql4-class-to-resource.php and removed all the decorate* functions from all the custom phtml files and it was done.

This projects uses algolia, a one page checkout and many modules from amasty, aheadworks and many other vendors, not the easiest and simplest setup.

wow

**Comment by luigifab** [comment]
A new age is coming!

**Reply by luigifab** [reply]
Beta2?

**Reply by fballiano** [reply]
i'm creating the release now

**Comment by djixas** [comment]
Does it include rwd2? Something [fballian](https://github.com/fballiano) talked about in "openmage future" post?
